python/ray/experimental/channel/serialization_context.py

```diff
@@ -0,0 +1,52 @@
+from typing import TYPE_CHECKING, List, Optional
+
+if TYPE_CHECKING:
+    import numpy as np
+    import torch
+
+
+class _SerializationContext:
+    def __init__(self):
+        self.torch_device: Optional["torch.device"] = None
+        self.use_external_transport: bool = False
+        self.tensors: List["torch.Tensor"] = []
+
+    def set_torch_device(self, torch_device: "torch.device") -> None:
+        self.torch_device = torch_device
+
+    def reset_tensors(self, tensors: List["torch.Tensor"]) -> List["torch.Tensor"]:
+        prev_tensors = self.tensors
+        self.tensors = tensors
+        return prev_tensors
+
+    def serialize_tensor(self, tensor: "torch.Tensor") -> "np.ndarray":
+        # Transfer through Ray's shared memory store for now.
+        # TODO(swang): This requires two copies, one to transfer from GPU to
+        # CPU and another from CPU to shared memory. Ideally we should elide
+        # the first copy and memcpy directly from GPU to the shared memory
+        # buffer.
+        if tensor.device.type == "cuda":
```

Comment:
Hmm I don't think `detach` matters yet since we only care about inference, but yes I think we will want to do that for training.

Not sure if there is much of a difference between `cpu()` vs `Tensor.to`.